lua/rustaceanvim/dap.lua

```diff
@@ -102,21 +103,80 @@ local function generate_source_map()
       local new_map = {
         [compat.joinpath('/rustc/', commit_hash)] = compat.joinpath(rustc_sysroot, '/lib/rustlib/src/rust'),
       }
-      vim.tbl_extend('force', source_map, { new_map })
+      source_map = vim.tbl_extend('force', source_map, { new_map })
     end)
   end)
 end
 
----@param args RADebuggableArgs
-function M.start(args)
-  vim.notify('Compiling a debug build for debugging. This might take some time...')
+---@type string[]
+local init_commands = {}
+
+local function get_lldb_commands()
+  get_rustc_sysroot(function(rustc_sysroot)
+    local script_import = 'command script import "' .. rustc_sysroot .. '/lib/rustlib/etc/lldb_lookup.py"'
+    local commands_file = rustc_sysroot .. '/lib/rustlib/etc/lldb_commands'
+    local file = io.open(commands_file, 'r')
+    if file then
+      for line in file:lines() do
+        table.insert(init_commands, line)
+      end
+      file:close()
+    end
+    table.insert(init_commands, 1, script_import)
+  end)
```

Comment:
potential issue: `init_commands` will grow whenever this function is called.
I would suggest changing `init_commmands` to a table (`---@type {[string]: string}`) where key = value, to avoid inserting duplicate entries.
Then we can use the keys or values as a list.

Reply:
Ah, right. Is maintaining these types of values in global scope the way forward, or would you prefer a refactor toward functions returning values?

Reply:
:thinking: 

The reason I use state is because the callback passed to `vim.system` is asynchronous in Neovim 0.10, so the source tables may not be ready by the time the user is presented with a choice of debuggables (but it should be by the time a choice is made). This makes the UI feel a bit more snappy (although I'll probably want to add some `await` calls before actually starting the debugger, to be on the safe side).

The way it's implemented right now - with only one state table - it's possible for multiple projects to interfere with each other. We should probably at least have a table with the workspace root as the keys to prevent this.